### PR TITLE
Sphere sculpting brush

### PR DESCRIPTION
**Pull request by msfeldstein**
If you want to make a snowman you have to be able to make spheres.  Adds a single sphere brush that you can make whatever size you want by dragging it out or in.
snowman

### CODE REVIEW

**Comment by msfeldstein** [comment]
Oops just realized i didn't add an icon thumbnail, i can add that when i get back to my mac

**Comment by feiss** [comment]
Really cool! I love this new way of using A-Painter to create single objects like the line or this sphere. Very simple and fun :)

I'd make some changes to this pr before merging:

* I'd use smoothShading, with frontSide only, and a less metallic appearance, more rubber-like.
```
        color: this.data.color,
        roughness: 0.6,
        metalness: 0.2,
        side: THREE.FrontSide,
        shading: THREE.SmoothShading
```
Example with this settings: http://imgur.com/a/THMGy

* Also, a 1m sphere appears in 0,0,0 when creating the first point. I'd make the `object3D` invisible at `init()` and then make it visible at second point.

* Lastly, set `spacing` to 0, so no there's no snapping and it has a much nicer feeling 

* ..and change thumbnail ;)

**Comment by feiss** [comment]
Something interesting with this 'new kind' of brushes is that despite only the first and last points would be really needed to get the final shape, all points are actually needed with the current save/load arquitecture. Cons: space wasted in scene  files. Pros: they will look great in playback mode :)

**Comment by msfeldstein** [comment]
Good feedback I'll update. Is there a playback mode? That would be huge for
non hmd development

**Comment by fernandojsg** [comment]
Really nice! I like the way to interact with it, simple and clever :+1: 
Regarding the feedback, I would change the name from `scuplt-sphere` to `single-sphere` or so, as we're not really sculpting here, maybe in the future if we add support for marching geometries or so we could go back to these naming.
As @feiss mentioned I should work on the save/load architecture to prevent store plenty of points for just one sphere, I'll take a look at it.
And re the playback mode, this is a feature I'm currently working on, I hope to have it ready before new year :)

**Comment by fernandojsg** [comment]
I would also change `THREE.SphereGeometry(0.5, 32, 32);` for something with less segments 32 by 32 generates a lot of faces probably to use it widely in a scene. Maybe an icosahedron with few levels of subdivision could do the work while keeping the same smooth aesthetics.

**Comment by feiss** [comment]
Oh yes, I forgot to mention that. It has to be more optimised indeed. I tried with SphereGeometry 16x16 and looked fine enough!

**Comment by msfeldstein** [comment]
Cool i think i hit all the points there. 
I switched to Icosahedron with level 2 detail which has 320 faces vs 480 faces for Sphere 16x16.  Level 3 detail is smoother but has 1280 faces.

**Comment by msfeldstein** [comment]
Also maybe in a follow up PR i can have it dynamically switch to a level 3 geometry if its a really huge sphere, while leaving it low LOD for small spheres.

File: src/brushes/single-sphere.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  {
+    init: function (color, width) {
+      this.material = new THREE.MeshStandardMaterial({
+        color: this.data.color,
+        roughness: 0.6,
+        metalness: 0.2,
+        side: THREE.FrontSide,
+        shading: THREE.SmoothShading
+      });
+      this.geometry = new THREE.IcosahedronGeometry(0.5, 2);
```
Comment (fernandojsg) [inline src/brushes/single-sphere.js]: Use`IcosahedronBufferGeometry`. It's always better to use the `BufferGeometry` version of each primitive.

File: src/brushes/single-sphere.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      this.object3D.add(this.mesh);
+      this.mesh.visible = false
+    },
+    addPoint: function (position, orientation, pointerPosition, pressure, timestamp) {
+      if (!this.firstPoint) {
+        this.firstPoint = pointerPosition.clone();
+        this.mesh.position.set(this.firstPoint.x, this.firstPoint.y, this.firstPoint.z)
+      }
+      this.mesh.visible = true
+      var distance = this.firstPoint.distanceTo(pointerPosition) * 2;
```
Comment (fernandojsg) [inline src/brushes/single-sphere.js]: Why don't change it to use unit sphere?
from:
```
this.geometry = new THREE.IcosahedronGeometry(0.5, 2);
var distance = this.firstPoint.distanceTo(pointerPosition) * 2;
```
to:
```
this.geometry = new THREE.IcosahedronGeometry(1, 2);
var distance = this.firstPoint.distanceTo(pointerPosition);
```

**Comment by msfeldstein** [comment]
Switched to using unit sphere, but the version of three packaged with this version of aframe doesn't have IcosahedronBufferGeometry yet.

**Comment by fernandojsg** [comment]
Aps, my fault :) we'll switch to them once we updated the version.
Thanks for the PR! it's looks great! :+1:

**Pull request merged**